quic/s2n-quic-transport/src/recovery/manager.rs

```diff
@@ -100,6 +100,13 @@ macro_rules! recovery_event {
     };
 }
 
+macro_rules! packet_info_type {
+    () => {
+        <<Config::CongestionControllerEndpoint as congestion_controller::Endpoint>::CongestionController as congestion_controller::CongestionController>::PacketInfo
+    }
+}
```

Comment:
can you add a comment explaining a little bit because it might be a bit surprising for new readers

Reply:
it should make you happy since there is no way you would attempt to do such a thing in C. It looks ugly, but the compiler is checking everything and it helps us save memory